source/macos/AdaptiveCards/AdaptiveCards/Managers/ACSVisibilityManager.swift

```diff
@@ -0,0 +1,167 @@
+//
+//  ACSVisibilityManager.swift
+//  AdaptiveCards
+//
+//  Created by uchauhan on 30/08/22.
+//
+//  This class is designed to handle the visibility of views as well as the corresponding padding or separator. Class also depended on Filler Space Manager to find associated views.
+
+import AdaptiveCards_bridge
+import AppKit
+
+/// VisibilityManager protocols for handle element visibility
+@objc protocol ACSIVisibilityManagerFacade {
+    func hideView(_ view: NSView)
+    func unhideView(_ view: NSView)
+}
+
+class ACSVisibilityManager {
+    /// tracks objects that are used in filling the space
+    private var fillerSpaceManager: ACSFillerSpaceManager
+    
+    /// tracks visible views
+    private var visibleViews: NSMutableOrderedSet
+    
+    var hasVisibleViews: Bool { return !visibleViews.set.isEmpty }
+    
+    init(_ fillerSpaceManager: ACSFillerSpaceManager) {
+        self.fillerSpaceManager = fillerSpaceManager
+        self.visibleViews = NSMutableOrderedSet()
```

Comment:
Is the plan to inject this later?
Because if not you can directly set this in the declaration

Reply:
let's discuss